File: python/tests/connect_test.py
import pytest
import dcss_api
from dcss_api import APIErr


def test_successful_connect():
    # Connect to DCSS Webtile
    webtile = dcss_api.WebtilePy("ws://localhost:8080/socket", 0, "0.29")

    assert '{"msg":"ping"}' == webtile.get_message()
    assert '{"msg":"lobby_clear"}' == webtile.get_message()
    assert '{"msg":"lobby_complete"}' == webtile.get_message()

    webtile.disconnect()


def test_failed_connect():
    try:
        webtile = dcss_api.WebtilePy("ws://localhost:XXXX/socket", 0, "0.29")
        assert False
    except APIErr as e:
        if "Url error" in e.args[0]:
            assert True
        else:
            assert False

    try:
        webtile = dcss_api.WebtilePy("ws://localhost:0000/socket", 0, "0.29")
        assert False
    except APIErr as e:
        if "Tungstenite error" in e.args[0]:
            assert True
        else:
            assert False

File: dcss-api/tests/connect_test.rs
use dcss_api::{Error, Webtile};
use serde_json::{Value, from_str};

#[test]
fn successful_connect() {
    let mut webtile =
        Webtile::connect("ws://localhost:8080/socket", 0).expect("Failed to connect.");

    let json: Value = from_str("{\"msg\":\"ping\"}").unwrap();
    assert_eq!(Some(json), webtile.get_message());
    let json: Value = from_str("{\"msg\":\"lobby_clear\"}").unwrap();
    assert_eq!(Some(json), webtile.get_message());
    let json: Value = from_str("{\"msg\":\"lobby_complete\"}").unwrap();
    assert_eq!(Some(json), webtile.get_message());
    assert_eq!(None, webtile.get_message());

    webtile.disconnect().expect("Failed to disconnect");
}

#[test]
fn failed_connect() {
    let webtile = Webtile::connect("ws://localhost:XXXX/socket", 0);
    assert!(matches!(*webtile.err().unwrap(), Error::Websocket(_)));

    let webtile = Webtile::connect("ws://localhost:0000/socket", 0);

    assert!(matches!(*webtile.err().unwrap(), Error::Websocket(_)));
}
